### PR TITLE
Limit steps for composing a tutorial stepper

### PR DESCRIPTION
**Pull request by freevie**
Motivation:

With the upcoming addition of gRPC tutorial, the stepper in the REST tutorial displays steps of the gRPC tutorial as well.

image

Modifications:

- Check the parent directory of the tutorial page and compose a stepper of sibling pages only.

Result:

- The stepper shows steps of the sibling pages only. (i.e. REST tutorial steps only on REST tutorial pages, gRPC tutorials steps only on gRPC tutorial pages)

### CODE REVIEW

File: site/src/components/steps.tsx [hunk 1]
```diff
@@ -28,20 +28,30 @@ const TutorialSteps: React.FC<TutorialStepProps> = (props) => {
...
+            ... on File {
+              relativeDirectory
+            }
+          }
         }
       }
     }
   `);
 
+  let tutorialType = location.pathname.substr(0, location.pathname.lastIndexOf("/"))
```
Comment (jrhee17) [inline site/src/components/steps.tsx:41]: Note: if someone happens to create a subdirectory under `/rest/blog` (`/rest/blog/inner/sample.mdx`), an independent `Steps` will be created. This behavior is undefined now anyways, so the current approach SGTM 👍

**Review by jrhee17 — APPROVED** [review]
Thank you! Looking forward to the grpc tutorial 🤩

**Comment by ikhoon** [comment]
Looking good. Could you check lint errors?
https://github.com/line/armeria/runs/4408067939?check_suite_focus=true#step:5:453

**Review by minwoox — APPROVED** [review]
Thank you! ❤️

**Comment by minwoox** [comment]
We also have this error
```
error "window" is not available during server side rendering.
failed Building static HTML for pages - 2.166s

  39 |   `);
  40 | 
> 41 |   const tutorialType = window.location.pathname
     |                        ^
  42 |     .substr(0, window.location.pathname.lastIndexOf('/'))
  43 |     .replace('/tutorials/', '');
  44 | 


  WebpackError:ReferenceError: window is not defined
```

I'm not a React expert but we can probably solve this using:
```
import React, { useState, useEffect } from 'react';

...

const useTutorialType = () => {
    const [tutorialType, setTutorialType] = useState("");

    useEffect(() => {
      setTutorial(window.location.pathname
                      .substr(0, window.location.pathname.lastIndexOf('/'))
                      .replace('/tutorials/', ''));
    });

    return tutorialType;
  }

  const tutorialSteps = Object.entries(
    tutorialNodes.filter(
      (tutorialNode) => tutorialNode.parent.relativeDirectory === useTutorialType(),
    ),
  ).map(...)
```

Could you try with this, please? 😄

**Comment by freevie** [comment]
@minwoox ,
Regarding https://github.com/line/armeria/pull/3970#issuecomment-987587701, it worked like a charm. Thank you for looking into this and am sorry for stealing your time.

I made a bit of modification to your suggestion, due to the following lint errors:

```
React Hook useEffect contains a call to 'setTutorialType'. Without a list of dependencies, this can lead to an infinite chain of updates. To fix this, pass [] as a second argument to the useEffect Hook  react-hooks/exhaustive-deps

React Hook "useTutorialType" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function                                                  react-hooks/rules-of-hooks
```

I've never had an experience with React before so I am not confident whether my modification is okay or not. Please see if the changes are okay.

**Comment by freevie** [comment]
I see that there is another error, but I can't seem to identify the cause of the error. Could you kindly direct me to where I should look into?

**Comment by minwoox** [comment]
> I see that there is another error, but I can't seem to identify the cause of the error. Could you kindly direct me to where I should look into?

It's a flaky test and you don't have to worry about it. 😄 

> Thank you for looking into this and am sorry for stealing your time.

No problem at all. 😉 

> I made a bit of modification to your suggestion,

That looks better. Thanks!

File: site/src/components/steps.tsx [hunk 2]
```diff
@@ -28,22 +28,39 @@ const TutorialSteps: React.FC<TutorialStepProps> = (props) => {
...
-  );
+  const [tutorialType, setTutorialType] = useState('');
+
+  useEffect(() => {
+    setTutorialType(
+      window.location.pathname
+        .substr(0, window.location.pathname.lastIndexOf('/'))
+        .replace('/tutorials/', ''),
+    );
+  }, []);
```
Comment (minwoox) [inline site/src/components/steps.tsx:49]: nit:
```suggestion
  });
```

File: site/src/components/steps.tsx [hunk 3]
```diff
@@ -28,22 +28,39 @@ const TutorialSteps: React.FC<TutorialStepProps> = (props) => {
...
-  );
+  const [tutorialType, setTutorialType] = useState('');
+
+  useEffect(() => {
+    setTutorialType(
+      window.location.pathname
+        .substr(0, window.location.pathname.lastIndexOf('/'))
+        .replace('/tutorials/', ''),
+    );
+  });
```
Comment (minwoox) [inline site/src/components/steps.tsx:49]:
```suggestion
  }, []);
```
Oops sorry about it. I was wrong on it. 😓

**Review by ikhoon — APPROVED** [review]
Thanks a lot, @freevie!

**Comment by minwoox** [comment]
Thanks! ❤️

**Pull request merged**